aws-predictions-tensorflow/src/main/java/com/amplifyframework/predictions/tensorflow/service/TensorFlowTextClassificationService.java

```diff
@@ -132,7 +148,8 @@ void classify(
         }
     }
 
-    private Sentiment fetchSentiment(String text) throws PredictionsException {
```

Comment:
If you already have package visibility for `classify(...)`, can you just test through it? 

The only problem I can see might be that latch, which would block stuff.

Ah, haha. Okay. Remember that crazy `Loadable<List<Loadable<?, PredictionsException>>, PredictionsException>`? If you use that, and then accept it as a dependency in the constructor, you can "fake" whether or not this service is loaded.

Assuming you can push `loaded.await();` into the implementation of that big Loadable, that'll help you when you run `classify` from test. (In source, you provide an impl that blocks. In test, you provide a `mock(Loadable.class)` that does whatever you want.

I guess the actual latch await call would change to:
```
if (!compoundLoadable.isLoaded()) {
    compountLoadable.load();
}
```
Or something like that.

Reply:
This test is to directly test the behavior of `fetchSentiment`. How about if I write more tests for blocking behavior in the future when I improve on the current latch design?